interpreter.py

```diff
@@ -198,6 +236,10 @@ def exec_operation(self, instruction):
             var_val = self.lookup_var(args[0])
             assert isinstance(var_val, Float)
             return Float(var_val.value)
+        elif opcode == LLVMZExt:
+            var_val = self.lookup_var(args[0])
+            assert isinstance(var_val, Integer)
+            return Integer(var_val.value)
```

Comment:
My suggestion is that `Integer` and `Ptr` should be immutable. It's fine if *some* values can be mutated, but not others. Typically this is done by convention. If you want -- but I don't think it's necessary -- you can make a hierarchy along the lines of:

```python
class Obj: pass

class MutObj(Obj): pass

class ImmutObj(Obj): pass

class Integer(ImmutObj): ...

class UserObj(Obj): ...
```
and so on.